### PR TITLE
feat(tag): add tag component

### PR DESCRIPTION
**Pull request by cal-smith**
closes the main part of #84

### CODE REVIEW

File: .storybook/experimental.component.ts [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+	EventEmitter
+} from "@angular/core";
 import { ExperimentalService } from "../src/experimental.module";
 
 @Component({
 	selector: "app-experimental-component",
 	template: ``
 })
 export class ExperimentalComponenent implements AfterViewInit, OnDestroy {
+
```
Comment (zvonimirfras) [inline .storybook/experimental.component.ts]:
```suggestion
```

File: src/tag/tag.component.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * Supported tags types for carbon v10
+ */
+export type TagTypeExperimental = "basic" | "red" | "magenta" | "purple" | "blue" | "cyan" | "teal" | "green" | "cool-gray" | "warm-gray";
+
+/**
+ * Component that represents a tag for labelling/categorizing using keywords
+ */
+@Component({
+	selector: "ibm-tag",
+	template: `<span><ng-content></ng-content></span>`
```
Comment (zvonimirfras) [inline src/tag/tag.component.ts]: Does it have to be in a span?

File: src/tag/tag.component.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+export class Tag {
+	/**
+	 * type of the tag determines the styling
+	 *
+	 * Reference `TagType` for v9 applications, and `TagTypeExperimental` for v10/v9 experimental mode applications
+	 */
+	@Input()
+	type: TagType | TagTypeExperimental = "ibm";
+
+	@HostBinding("attr.class") get attrClass() {
```
Comment (zvonimirfras) [inline src/tag/tag.component.ts]: Should we allow setting of additional classes?

File: src/tag/tag.component.ts (hunk 2 above)
Comment (cal-smith) [reply src/tag/tag.component.ts]: nope, since it's applying to the host

File: src/tag/tag.component.ts (hunk 3 above)
Comment (cal-smith) [reply src/tag/tag.component.ts]: Why not

**Pull request merged**